### ISSUE TITLE
TypeScript do not detect erroneous intersection type declarations

### ISSUE THREAD

**Issue by maxloh** [microsoft/TypeScript#43862]
# Bug Report

### 🔎 Search Terms

- Intersection

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about intersection

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/dev/bug-workbench/?#code/C4TwDgpgBAYg9nKBeKBvKBDANmAFhgIwmAC4oAiDcqAHwoOrvIGNyAaKAEwEsBzb0lAB2AVwC2RAE5QAvlABkUABQAoKLTSYc+IoMrsufAWQCMstRvTY8hYmXIMOPfoIBM59XSvbbe1k6NBAGYPSy0bXXtOA2djKAAWcwBKAG4VFWY4IQBnYCgAMwQyeEQUbwi7CmiAlzJEmTSgA)

### 💻 Code

```ts
type Foo = { alphabet: "a" | "b" | "c", digit: number } & (
  | { alphabet: "a", digit: 1 }
  | { alphabet: "b", digit: 2 }
  | { alphabet: "c", digit: 3 }
  | { alphabet: "d", digit: 4 }
); // TypeScript should report error on this line, as "d" is not assignable to type "a" | "b" | "c"

const foo: Foo = { alphabet: "d", digit: 4 }; // But TypeScript didn't report error until this line
```

### 🙁 Actual behavior

TypeScript only report error when we assign an erroneous value to the erroneous type.

### 🙂 Expected behavior

TypeScript report error on type declaration.

**Comment by RyanCavanaugh** [comment]
Instantiations such as type intersections can't create errors, by design. These sorts of "impossible" intersections get routinely created during generic instantiation; there isn't really a clean place where we can safely error on them without creating incorrect errors in higher-order constructs.
